### PR TITLE
[MNG-7342] Add a --very-verbose / -XX cli option

### PR DESCRIPTION
**Pull request by gnodet**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by rmannibucau** [comment]
no strong opposition  but shouldnt it be superseeded by the CLI configuration of loggers? (kind of drop specific level config for a generic CLI solution, don't recall if we decided to go with --logger or something else but sounds better no?)

File: maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java [hunk 1]
```diff
@@ -48,6 +48,8 @@
 
     public static final char VERBOSE = 'X';
 
+    public static final String VERY_VERBOSE = "XX";
```
Comment (rfscholte) [inline maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java:51]: If this is really really required, please rename it to TRACE, IMO that's the common way to call it.

**Comment by gnodet** [comment]
I'm not sure about this PR actually.  My original problem was that Sisu does log exceptions during bean instantiation to TRACE level, which means it's not available when using the usual `-X` option.  This PR is only a workaround imho, and I'd much prefer having Sisu logging at a slightly higher level.   The problem with logging at TRACE level is that is that you have no way to be aware about the problem, which I think is wrong: if a class is missing, a ClassNotFoundError is thrown, caught and logged at TRACE level, but this looks kinda hidden to me.

**Comment by rmannibucau** [comment]
shouldnt -X enable trace? -X is a kind of enable them all IMHO (otherwise what is its goalas "this"?)

**Comment by gnodet** [comment]
> shouldnt -X enable trace? -X is a kind of enable them all IMHO (otherwise what is its goalas "this"?)

That was my original suggestion in https://issues.apache.org/jira/browse/MNG-7342

**Comment by rmannibucau** [comment]
@gnodet think it makes sense, verbose is already unusable if you dont tune with -D so let's just make it "ALL", no real reason to fear trace will be more verbose, we are already far behind being acceptable without grep/search ;)

**Comment by gnodet** [comment]
> @gnodet think it makes sense, verbose is already unusable if you dont tune with -D so let's just make it "ALL", no real reason to fear trace will be more verbose, we are already far behind being acceptable without grep/search ;)

I do agree. @michael-o  ?

**Comment by rfscholte** [comment]
`-Dorg.slf4j.simpleLogger.log.Sisu=trace` didn't work? If this does work (or something similar) I wouldn't add a new flag just for simplifying the edge case.

**Comment by gnodet** [comment]
> `-Dorg.slf4j.simpleLogger.log.Sisu=trace` (as system property, e.g. via MAVEN_OPTS) didn't work? If this does work (or something similar) I wouldn't add a new flag just for simplifying the edge case.

It does I suppose.  
The problem wasn't really in enabling the logging, rather that, at the time I could understand that I had to raise the logging, I already had fully analysed / understood what the problem was and how to fix it, i.e. find that the problem comes from the injection, debugging the Sisu code, understand where the problem is located and find out I could have raised the log level.  
I expected the `-X` option to show any problem, but it did not, hence my original proposal to raise `-X` to display `TRACE` level.  After this solution, I'd favour fixing https://github.com/eclipse/sisu.inject/issues/36, then this PR if none of the previous solution is acceptable.

File: maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java (hunk 1 above)
Comment (michael-o) [reply maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java:51]: @rfscholte That would be against the logic we have deprecated `--debug` with `-X` only.

**Comment by michael-o** [comment]
> > @gnodet think it makes sense, verbose is already unusable if you dont tune with -D so let's just make it "ALL", no real reason to fear trace will be more verbose, we are already far behind being acceptable without grep/search ;)
> 
> I do agree. @michael-o ?

I don't how much we really need `TRACE` support out of the box. Given that @rfscholte provided a solution for this specific problem and that --logger {logger}:{level} would be a better way to solve. I am split here. The benefit isn't too big to justify this.

**Comment by cstamas** [comment]
From [sisu doco](https://www.eclipse.org/sisu/docs/tutorials/index.html#go): "If you're interested in what Sisu is doing you can turn on tracing with -Dsisu.debug"

File: maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java (hunk 1 above)
Comment (rfscholte) [reply maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java:51]: Yes, I see your point, although has "tracing" multiple meanings, like debug does?

**Comment by gnodet** [comment]
> From [sisu doco](https://www.eclipse.org/sisu/docs/tutorials/index.html#go): "If you're interested in what Sisu is doing you can turn on tracing with -Dsisu.debug"

Again, the problem is not to find a way to raise sisu log.  The problem was to actually find what the problem was.  I had no idea that the problem came from sisu at all and I could not imagine that a NoClassDefFoundError was simply logged at _trace_ level.  Trace is not meant to display exceptions, but rather very fine grained info.  Such problems should at least be logged at _debug_ level, hence appearing with the standard `-X` option.  Anything useful should at least happen here.

I also faced a problem recently where an extension was not working in some environment.  I kinda abandoned debugging it (because debugging was not possible), but I now suspect that it was the same kind of issue which was not printed at debug level.

I do agree we have tons of ways to raise specific loggers level.  The goal here is to have a simple option that will at least display possible problems without having to know which log level you need to raise or even that you need to raise one.  Any suggestion that provides such a workaround relies on the user understanding where the problem is, which is what I'm trying to help him finding.

File: maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java (hunk 1 above)
Comment (michael-o) [reply maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java:51]: Correct and we should avoid multiple meanings, no?

**Comment by michael-o** [comment]
@gnodet I do understand the problem and I think it must be solved in Sisu, as noted in the Sisu ticket. The messages are in the wrong log level.

**Comment by gnodet** [comment]
The sisu PR has been merged, closing this one.

**Comment by jira-importer** [comment]
Resolve #8081

**Pull request closed without merging**